File: MeteoriteWeb/src/app/interfaces/meteorite-filter.ts
export interface MeteoriteFilter {
    year?: YearFilter,
    recClass?: string,
    name?: string,
    sorting?: Sorting,
}

export interface YearFilter {
    startDateYear?: number,
    endDateYear?: number,
}

export interface Sorting {
    sortBy?: SortBy,
    sortOrder?: SortOrder,
}

export enum SortBy { Year, Count, SumMass, };
export enum SortOrder { Ascending, Descending, };

File: MeteoriteAPI/MeteoriteAPI.Common/Models/Filters/MeteoriteFilter.cs
namespace MeteoriteAPI.Common.Models.Filters
{
    public class MeteoriteFilter
    {
        public YearFilter? Year { get; set; }

        public string? RecClass { get; set; }

        public string? Name { get; set; }

        public Sorting? Sorting { get; set; }
    }
}
